src/pages/SignUp/SignUp.js

```diff
@@ -1,10 +1,197 @@
 import React, { Component } from 'react';
+import './SignUp.scss';
 
 class SignUp extends Component {
+    constructor() {
+        super();
+        this.state = {
+            id: '',
+            pw: '',
+            name: '',
+            phone:'',
+            email: '',
+            passwordcheck: '',
+
+            validID: true,
+            validPW: true,
+            validDoubleCheck: true,
+            validName: true,
+            validPhone: true,
+            validEmail: true,
+        }
+    }
+    validationCheck = (type, value) => {
+        if (type==='id') {
+            let idReg = /^[a-z]+[a-z0-9]{5,19}$/g;
+            const validID = idReg.test(value);
+            this.setState({validID});
+        } else if (type==='pw') {
+            let passwordRules = /^(?=.*[a-zA-Z])(?=.*[!@#$%^*+=-])(?=.*[0-9]).{8,16}$/;
+            const validPW = passwordRules.test(value);
+            this.setState({validPW})
+        } else if (type === 'name') {
+            let reg = /^[가-힣]{2,4}$/;
+            const validName = reg.test(value);
+            this.setState({validName});
+        } else if (type === 'email') {
+            let emailRule = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+            const validEmail = emailRule.test(value);
+            this.setState({validEmail})
+        }
+    }
+
+    handleInputValueChange = (e) => {
+        // const { id, pw, name, phone, email } = e.target;
+        this.validationCheck(e.target.name, e.target.value);
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    checkValidation = (e) => {
+        e.preventDefault();
+        // console.log('연결확인');
+    
+        const {id, pw, name, phone, email, passwordcheck} = this.state;
+    
+        fetch('http://10.58.4.236:8000/user/signup', {
+          method: "POST",
+          body: JSON.stringify({
+            ID: id, 
+            password: pw,
+            name: name,
+            phone_number: phone,
+            email: email,
+            passwordcheck: passwordcheck,
+          }),
+        })
+          .then((res) => res.json())
+          .then((result) => {
+
+        if (result.message === "success") {
+          alert ('회원가입이 완료되었습니다!')
+          this.props.history.push("/Login")
+        } else {
+          alert(result.message);
+        }})
```

Comment:
```suggestion
        if (result.message === "success") {
          alert ('회원가입이 완료되었습니다!')
          this.props.history.push("/Login")
        } 
         return alert(result.message);
        })
```
return을 작성해주면 else문이 불필요할 것 같네요!

Reply:
수정했습니당:)